source/userguide/dm.rst

```diff
@@ -659,6 +661,106 @@ For the use case of transferring money in a banking assistant application, the f
                    "{account_from} to a {account_to}. Is that right?"]
         responder.reply(replies)
 
+
+Subforms and invoking slot-filling within a dialogue handler
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To support dynamic iteration through a form, we suggest the use of additional subforms or follow-up forms. Consider a case in which the initial form fetches a certain set of entities, and the flow ahead depends on the values of those captured entities. This requires capturing of further entities based on those values. For this, we recommend using further subforms.
+
+Take into consideration the previous example. Now, instead of a single pass form to capture all entities, say we want to check whether the account from which money is withdrawn is a savings account or not. If yes, then continue to fetch the other entities (using follow-up form ``form_transfermoney_2``), otherwise send appropriate response.
+
+.. code:: python
+
+    from mindmeld.components.dialogue import AutoEntityFilling
+    from mindmeld.core import FormEntity
+
+    form_transfermoney_1 = {
+        'entities':[
+            FormEntity(
+                entity='account_type',
+                role='account_from',
+                responses=['Sure. Transfer from which account?']
+                ),
```

Comment:
hints=['checking', 'checkings']

Reply:
No, this one is without any hints. The check happens inside the dialogue handler for 'savings', if you go through the flow.